AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java

```
@@ -206,10 +210,14 @@ public static void cancelTask() {
     }
 
 
-    public static void cancelTask(TASK_TYPE taskType) {
-        // cancel the current task only if it's of type taskType
-        if (sLatestInstance != null && sLatestInstance.mType == taskType) {
-            cancelTask();
+    /** Cancel all tasks of type taskType*/
+    public static void cancelAllTasks(TASK_TYPE taskType) {
+        synchronized (sTasks) {
+            for (CollectionTask task: sTasks) {
+                if (task.mType == taskType) {
+                    task.cancel(true);
```

The closer I look at it the more little things I see, sorry. But, cancelTask(TASK_TYPE) used to delegate to cancelTask, which just silently eats all Exceptions. Now you are cancelling the task directly so it's missing all this safety code:

```java
        try {
            if ((sLatestInstance != null) && (sLatestInstance.getStatus() != AsyncTask.Status.FINISHED)) {
                sLatestInstance.cancel(true);
                Timber.i("Cancelled task %s", sLatestInstance.mType);
            }
        } catch (Exception e) {
            return;
        }
```

- the finished check
- the logging that it's being cancelled
- the exception catch

---

I added a "safeCancel" which cancel the object, and call it from cancelTask and cancelAllTasks